eva_submission/steps/rename_contigs_from_insdc_in_assembly.py

```diff
@@ -0,0 +1,158 @@
+        self.assembly_accession = assembly_accession
+        self.assembly_fasta_path = assembly_fasta_path
+        self.assembly_report_path = assembly_report_path
+
+    @cached_property
+    def contigs_found_in_vcf(self):
+        """Provides the contigs present in the VCF file"""
+        contigs = set()
+        for input_vcf in self.input_vcfs:
+            if input_vcf.endswith('.gz'):
+                vcf_in = gzip.open(input_vcf, mode="rt")
+            else:
+                vcf_in = open(input_vcf, mode="r")
+            for line in vcf_in:
+                if line.startswith("#"):
+                    continue
+                contigs.add(line.split('\t')[0])
+        return contigs
+
+    @staticmethod
+    def _get_assembly_report(assembly_report):
+        """Parse the assembly report and return each row as a dict."""
+        headers = None
+        with open(assembly_report) as open_file:
+            # Parse the assembly report file to find the header then stop
+            for line in open_file:
+                if line.lower().startswith("# sequence-name") and "sequence-role" in line.lower():
+                    headers = line.strip().split('\t')
+                    break
+            reader = DictReader(open_file, fieldnames=headers, dialect=excel_tab)
+            return headers, [record for record in reader]
+
+    @cached_property
+    def assembly_report_rows(self):
+        """Provides assembly report rows with each row as a dict."""
+        headers, rows = self._get_assembly_report(self.assembly_report_path)
+        self.assembly_report_headers = headers
+        return rows
+
+    @cached_property
+    def assembly_report_map(self):
+        """
+        Dictionary of INSDC accession to naming convention used in the VCF constructed based on the assembly report.
+        """
+        assembly_report_map = {}
+        for row in self.assembly_report_rows:
+            # Search if the contig name is found in the VCF
+            for header in ['# Sequence-Name', 'GenBank-Accn', 'RefSeq-Accn', 'UCSC-style-name']:
+                if row[header] in self.contigs_found_in_vcf:
+                    assembly_report_map[row['GenBank-Accn']] = row[header]
+        return assembly_report_map
+
+    @retry(tries=3, delay=2, backoff=1.2, jitter=(1, 3))
+    def _contig_alias_assembly_get(self, page=0, size=10):
+        """queries the contig alias to retrieve the list of chromosome associated with the assembly"""
+        url = (f'https://www.ebi.ac.uk/eva/webservices/contig-alias/v1/assemblies/{self.assembly_accession}/'
+               f'chromosomes?page={page}&size={size}')
+        response = requests.get(url, headers={'accept': 'application/json'})
+        response.raise_for_status()
+        response_json = response.json()
+        return response_json
+
+    @staticmethod
+    def _add_chromosomes_to_map(assembly_data, contig_alias_map):
+        """Add entries to a map based on the contig alias response."""
```

Comment:
Maybe in the method name or this docstring, mention this map is from alternative naming convention to INSDC accession (as opposed to `contig_alias_map` which is from INSDC to the alternative).

Reply:
Tried to make it clearer.